docs/DESIGN.md

```diff
@@ -0,0 +1,77 @@
+# YesNo Design
+
+This document is meant to describe the guiding principles and direction of yesno development. When adding new features or prioritizing the backlog, these ideas should help in decision making.
+
+## Usage Types
+
+1. `yesno.save, yesno.load`
+1. `yesno.record, yesno.complete`
+1. `yesno.test`
+
+Case 1 is the most basic and requires the most work by the user. It is intended as an escape hatch for users that have unique or specific needs that the default usage doesn't meet. Most users should be using cases 2 and 3 (where case 3 is just a convience testing wrapper around case 2).
```

Comment:
Can you include some of the thoughts around competing with Nock. Maybe add a subsection here ### Differentiating from Nock

And call out that usecase 1 is very nock like and yesno can't outcompete nock here. Use case 2-3 yes nockback which isn't that great and we will do better and differentiate here. 

That more or less.

Reply:
added

Reply:
Looks good. Can you just call out nockback as a competing feature? I think it's helpful

Reply:
Ok, I added a sentence on nockback